clang/test/CXX/temp/temp.decls/temp.variadic/p5.cpp

```diff
@@ -369,9 +369,39 @@ void test_unexpanded_exprs(Types ...values) {
   void f(int arg = values); // expected-error{{default argument contains unexpanded parameter pack 'values'}}
 }
 
-// Test unexpanded parameter packs in partial specializations.
-template<typename ...Types>
-struct TestUnexpandedDecls<int, Types>; // expected-error{{partial specialization contains unexpanded parameter pack 'Types'}}
+// Test unexpanded parameter packs in partial/explicit specializations.
+namespace Specializations {
```

Comment:
This test is not showing any change in behavior? All of these diagnostics are existing behavior (https://godbolt.org/z/xr49ac7Ed). What actual changes does this patch make?

Reply:
@erichkeane The existing behavior is to call explicit specializations "partial specializations" in the diagnostic for unexpanded parameter packs. This patch fixes that, so a distinction is made between partial/explicit specializations. For example, the test on line 391 displays different behavior:
```cpp
struct InnerClass<Ts>; // expected-error{{explicit specialization contains unexpanded parameter pack 'Ts'}}
```
Note "explicit specialization" in the diagnostic.